src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js

```diff
@@ -95,36 +99,35 @@ export class RuleEditor extends LinkedComponent {
       this.ruleNameLink,
       this.deviceGroupLink,
       this.conditionsLink,
+      this.timePeriodLink,
+      this.calculationLink
     ].every(link => !link.error);
   }
 
   apply = (event) => {
     event.preventDefault();
     const { formData } = this.state;
     const { onClose, insertRule, updateRule } = this.props;
-    console.log('TODO: Handle the form submission');
+    if (formData.calculation === calculations[1]) this.setState({ formData: { ...this.state.formData, timePeriod: '' } })
```

Comment:
No need to update the state here. This could cause some async issues. Just update a local version of the state formData.